napari/layers/shapes/shapes.py

```diff
@@ -2642,18 +2643,32 @@ def _finish_drawing(self, event=None) -> None:
                     self._data_view.edit(index, vertices[:-1])
             if self._mode in {Mode.ADD_POLYGON, Mode.ADD_POLYGON_LASSO}:
                 vertices = self._data_view.shapes[index].data
+                if self._mode == Mode.ADD_POLYGON_LASSO:
+                    prev_vertices = len(vertices)
+                    vertices = rdp(
+                        vertices,
+                        epsilon=get_settings().experimental.rdp_epsilon,
+                    )
+                    if len(vertices) <= 3 and prev_vertices > 3:
+                        # https://github.com/napari/napari/issues/7903
+                        show_warning(
+                            trans._(
+                                'Polygons must have three or more vertices. '
+                                'Lasso polygons are simplified using an '
+                                'algorithm called RDP, which may cause '
+                                'polygons smaller than the RDP epsilon '
+                                'parameter to disappear entirely. If you are '
+                                'having trouble drawing small polygons, try '
+                                'reducing the value of napari > Settings > '
+                                'Experimental > RDP epsilon.'
```

Comment:
```suggestion
                                'Polygons must have three or more vertices. '
                                'Lasso polygons are simplified using the '
                                'RDP algorithm, which may cause polygons '
                                'smaller than RDP epsilon to disappear. If  '
                                'you face issues drawing small polygons, '
                                'try reducing napari > Settings > '
                                'Experimental > RDP epsilon. '
```
![image](https://github.com/user-attachments/assets/0b165cae-ebff-4455-a8a3-77e43553cfae)

This works for me